core/unit_test/TestTeamTeamSize.hpp

```diff
@@ -53,8 +53,20 @@ struct FunctorReduce {
 using policy_type = Kokkos::TeamPolicy<TEST_EXECSPACE>;
 using policy_type_128_8 =
     Kokkos::TeamPolicy<TEST_EXECSPACE, Kokkos::LaunchBounds<128, 8> >;
+
+// We need to special case for NVIDIA architectures which don't have space for
+// 2048 threads ptxas warns and with errors as warning errors out: "ptxas error
+// : Value of threads per SM for entry _ZN6... is out of range. .minnctapersm
+// will be ignored" And yes I understand I am lying now with the name of the
+// policy
```

Comment:
Just as a comment: I did check that this warning/error happens when compiling for Amper86 for example.